### ISSUE TITLE
Port Mypal to Ubuntu 8.04

### ISSUE THREAD

**Issue by brunochanrio** [Feodor2/Mypal68#361]
It's possible to port Mypal to Ubuntu 8.04 (and also to other old Linux operating systems)?

**Comment by Feodor2** [comment]
Don't know.
But i can give you tips
1. The rust - 1.45 version, check if its possible to run it then compile any program and run that program.
2. The cpp compiler of 17 level standard, clang-cl 9 used on windows but i don't know what is used on the linuх, supposed any 17 level, do the same checks as above

If these thing is good, then you may try to build

**Comment by skipster1337** [comment]
I tried cloning the github repo and running mach compile but i think it needs python2 because it was complaining about missing __builtin__ module which got changed to builtins in python3

https://stackoverflow.com/questions/9047745/where-is-the-builtin-module-in-python3-why-was-it-renamed

I was running debian 12 in a vm and the last debian version that had python2 was 11, plus i didn't have enough hard drive space for the compilation anyway

There is also a thing that old firefox versions don't work anymore in newer linux, i tried firefox 78 esr binary from mozilla's ftp on arch like two years ago and every single tab you'd open would crash instantly, it worked on debian for a while until a recent update. I think it's some dependency in libc but I'm not competent enough to answer on that. I imagine Mypal would have the same problem if it was compiled on linux. But it would be interesting to try.

**Comment by skipster1337** [comment]
My point is it would be far more realistic to get it compiled on modern linux first and then try to backport it to older versions

**Comment by skipster1337** [comment]
I tried again, this time on arch, I got python2 from the aur and got farther in the configuration process only to get told I don't have a version of rust installed even though it's right there. That's... interesting.
![2024-04-06 21_42_14-Window](https://github.com/Feodor2/Mypal68/assets/91155070/5ae3f834-af31-4353-b809-005659c29a77)

**Comment by skipster1337** [comment]
I got a bit farther now but for some reason it wants a directx sdk even though i am targeting amd64-linux in mozconfig. IDK what to edit in the build scripts to get past that. But yeah it's not gonna be very easy to adapt Mypal to build on linux.
![2024-04-06 23_10_47-Window](https://github.com/Feodor2/Mypal68/assets/91155070/b42bffb1-e5d7-4bab-8495-d556e0ec0888)

**Comment by skipster1337** [comment]
The solution was to remove that part from the build config, i managed to get a bit farther by commenting out xp-specific build commands but now it's complaining about some checksum mismatch where the check was put in place by the stubborn rust devs to make it impossible to continue without "forking" the code properly. This is about as far as i'm going to go with this, maybe someone smarter than me could figure out how to get past it.
![2024-04-07 00_35_23-Window](https://github.com/Feodor2/Mypal68/assets/91155070/6e984a56-d7ab-46de-8868-fd7b01b1309b)
![2024-04-07 00_35_46-Window](https://github.com/Feodor2/Mypal68/assets/91155070/f0da7dc2-4716-4bee-ae32-967fe5897dcb)

**Comment by NS-Clone** [comment]
> I hate the checksum shit too, i think its unnecessary here. This may be about the whitespace thing, rust files have many of them, and for checksum it is to be preserved. You may also delete bad checksums inside json files which lay in the rust components folders.

а можно по pуcскe чего там не так?

**Comment by Feodor2** [comment]
Чо скучно что-ли,  поверь ничего интересного для непрограмиста,  и теи более если словарик пробвал и не поенял.
А @skipster1337 если больше вопросов не задаёт всё понял и давно себе сделал под линкух версию.

**Comment by NS-Clone** [comment]
> Чо скучно что-ли, поверь ничего интересного для непрограмиста, и теи более если словарик пробвал и не поенял. 

а кто сказал что я не пограмизд? о_О
я и спрашиваю что вдруг пригодитсо
ибо такого еще не видел
и штуки 4...5 софтов под ХРю обратно я ужо подогнал

а словарик мне и не нужон
а лог таких масштабов без поиска и нормальной подсветки осилить глазами не реально
чтоб понять про какие чексуммы в коде? идет речь

**Comment by Feodor2** [comment]
Ладно я сборку запустил и можно потрындеть
Так что тебе не понятно? У него видно с первых строк всё чётко
А вот смотри как у меня круто
![image](https://github.com/Feodor2/Mypal68/assets/19492771/34ed571e-466d-4ff6-80d3-33ede2048677)

**Comment by skipster1337** [comment]
> А @skipster1337 если больше вопросов не задаёт всё понял и давно себе сделал под линкух версию.

No, I am just busy. I haven't done anything new yet, will try to remove the checks soon.

**Comment by skipster1337** [comment]
> > I hate the checksum shit too, i think its unnecessary here. This may be about the whitespace thing, rust files have many of them, and for checksum it is to be preserved. You may also delete bad checksums inside json files which lay in the rust components folders.
> 
> а можно по pуcскe чего там не так?

Там контрольные суммы в rust файлах которые нужно особым образом редактировать чтобы на них не ругался компилятор (или я так понял). Вроде их можно просто удалить или заменить на новые. Я ещё не попробовал, потому что занят.

**Comment by skipster1337** [comment]
i changed the checksum in the rust file but now i'm getting unknown variables and this permission error with the linker (running with sudo doesnt fix it)
![2024-04-17 15_56_37-Greenshot](https://github.com/Feodor2/Mypal68/assets/91155070/dbb54a7f-ba50-4cdd-905b-d32c26202ae1)
![2024-04-17 15_56_05-Greenshot](https://github.com/Feodor2/Mypal68/assets/91155070/4fbe5812-89de-4825-bdd3-7c69af2a998e)
![2024-04-17 15_55_56-Greenshot](https://github.com/Feodor2/Mypal68/assets/91155070/70237a02-4336-4978-8717-be85cfc32f12)
![2024-04-17 15_55_17-Greenshot](https://github.com/Feodor2/Mypal68/assets/91155070/feb24540-523d-45b2-86a6-6d4cb2f788b4)

**Comment by NS-Clone** [comment]
> > > I hate the checksum shit too, i think its unnecessary here. This may be about the whitespace thing, rust files have many of them, and for checksum it is to be preserved. You may also delete bad checksums inside json files which lay in the rust components folders.
> > 
> > 
> > а можно по pуcскe чего там не так?
> 
> Там контрольные суммы в rust файлах которые нужно особым образом редактировать чтобы на них не ругался компилятор (или я так понял). Вроде их можно просто удалить или заменить на новые. Я ещё не попробовал, потому что занят.

шо в этом будуйщем копирастического опенсорса
еще и "открытые коды" подписанны? :rofl: 👏

**Comment by Feodor2** [comment]
Hey next please post in by text and collapsed, pictures are really not convenient.
So nothing ti say about permission error
Next put #include "jsfriendapi.h" to that file
Then mutex file: may be I had broken it for linux when fixed for winxp, take it from the original firefox68 code

**Comment by skipster1337** [comment]
i managed to fix the jsfriendapi error but replacing the mutex file with the original ff68 version didn't help, i get the same error

**Comment by skipster1337** [comment]
also it looks like the permission errors fixed themselves, i realized i was using rustc 1.77 instead of 1.45

**Comment by skipster1337** [comment]
well now this stupid build system is telling me that the checksum for mod.rs is wrong but if i set it to its 'actual' checksum it does some weird shit and switches the values around and then tells me the 'actual' checksum value now is the 'expected' one from before, resulting in an endless loop
i honestly have no idea what the fuck it wants
![VirtualBox_arch_25_04_2024_18_59_55](https://github.com/Feodor2/Mypal68/assets/91155070/0736f8e8-52bd-422b-9ad6-74b82e3530fc)

theres more errors too
[buildlog.txt](https://github.com/Feodor2/Mypal68/files/15112764/buildlog.txt)

**Comment by Feodor2** [comment]
> replacing the mutex file with the original ff68 version didn't help

Try replace all that files with the original h cpp, may be the whole dir

> the checksum for mod.rs is wrong

remove it with the checksum from the json file, that whole string

**Comment by Feodor2** [comment]
Pay attention that remove json does not work too. You a are delete checksimse but leave the file.
{"files":{},"package":"xxxxx"} leave the package, seems it harmless.

Will be good to remove checksum shit completely, but no time to invest, as the shit laid seldom.

**Comment by K4sum1** [comment]
> well now this stupid build system is telling me that the checksum for mod.rs is wrong but if i set it to its 'actual' checksum it does some weird shit and switches the values around and then tells me the 'actual' checksum value now is the 'expected' one from before, resulting in an endless loop i honestly have no idea what the fuck it wants, i dont think it edits the rs file itself because the last modified date is from 3 weeks ago ![VirtualBox_arch_25_04_2024_18_59_55](https://github.com/Feodor2/Mypal68/assets/91155070/0736f8e8-52bd-422b-9ad6-74b82e3530fc)
> 
> theres more errors too [buildlog.txt](https://github.com/Feodor2/Mypal68/files/15112764/buildlog.txt)

I know it's been over a year and idk if this will be worked on again, but for r3dfox I ran into this issue since I had to modify the .rs files, and this was my solution.

https://github.com/Eclipse-Community/eCargo/commit/8c381008318f9e086af9192d0c03c03373078f20
